tests/unittests/unittests.cpp

```diff
@@ -60,6 +73,11 @@ class ReaderWriterQueueTests : public TestClass<ReaderWriterQueueTests>
 		REGISTER_TEST(threaded);
 		REGISTER_TEST(blocking);
 		REGISTER_TEST(vector);
+#if MOODYCAMEL_HAS_EMPLACE
+		REGISTER_TEST(emplace);
+		REGISTER_TEST(try_enqueue_bad);
```

Comment:
Not sure what this one is testing, exactly? It references `try_emplace` in a comment but doesn't use it.

Reply:
It's demonstrating what you have to do to try_enqueue() and movable type, which works but is ugly/slow, i.e. this is what try_emplace() is fixing.